### ISSUE TITLE
After loading the saved model it does not give the same results as the one in memory

### ISSUE THREAD

**Issue by pedrofernandoch** [keras-team/keras-io#1309]
I am following the [Image similarity estimation using a Siamese Network with a triplet loss](https://github.com/keras-team/keras-io/blob/master/examples/vision/siamese_network.py#L340) example.

I see the model is trained [here](https://github.com/keras-team/keras-io/blob/master/examples/vision/siamese_network.py#L346).

siamese_model.fit(train_dataset, epochs=10, validation_data=val_dataset) 

Now, I'm trying to save the model like this:
tf.saved_model.save(embedding, 'content/saved_model/')

When I do this, i receive warnings like these "WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
WARNING:absl:Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 53). These functions will not be directly callable after loading."

After this I loaded the model like this:
loaded_model = tf.keras.models.load_model('/content/saved_model')

The problem is that the model that is still in memory (embedding) and the model loaded (loaded_model ) produces different embeddings for the same images, the one is memory is much better and produces good results, the one that was loaded is trash.

What am I doing wrong? I need to store this model so I can use it later by loading, because it does not make any sense to train the model again every time I need it.

**Comment by dhantule** [comment]
Hi @pedrofernandoch, thanks for reporting this. 

I have tested this example and I'm able to save and load the model using 

```
embedding.save('saved_model.keras') 
loaded_model = keras.models.load_model('saved_model.keras')
```

Please refer this attached [gist](https://colab.sandbox.google.com/gist/dhantule/ddfcf3402eca89892866de27f2c9a54a/copy-of-siamese_network.ipynb), let me know if you still face any issues. Thanks!
